### PR TITLE
Option to read metadata from a JSON file

### PR DESCRIPTION
**Pull request by digitalorder**
This change adds an option to provide user-defined metadata from a file which is expected to have a valid JSON inside.

There is already a way to give JSON as a string (`--metadata-from-json`) parameter, but it requires presence of both `'` and `"` indicators. Now if one would like to pass such metadata string through multiple shell scripts and utilities all of them would have to respect quotations carefully before they land in eventual `pytest` call. With a JSON file this problem would just go away given the file has no spaces inside the name.

### CODE REVIEW

**Comment by BeyondEvil** [comment]
> This change adds an option to provide user-defined metadata from a file which is expected to have a valid JSON inside.
> 
> There is already a way to give JSON as a string (`--metadata-from-json`) parameter, but it requires presence of both `'` and `"` indicators. Now if one would like to pass such metadata string through multiple shell scripts and utilities all of them would have to respect quotations carefully before they land in eventual `pytest` call. With a JSON file this problem would just go away given the file has no spaces inside the name.

This is great!

During the holidays I'm going to fix CI so we can get this merged!

File: pytest_metadata/plugin.py
Comment (ImXron) [inline pytest_metadata/plugin.py:96]: You should be able to do:

```python
config._metadata.update(json.load(json_file))
```

File: tests/test_metadata.py
Comment (ImXron) [inline tests/test_metadata.py:112]: YES!

File: pytest_metadata/plugin.py
Comment (digitalorder) [reply pytest_metadata/plugin.py]: Indeed. This should be more readable and faster. Fixed.

**Review by ImXron — DISMISSED** [review]
LGTM! @BeyondEvil need any help with the CI stuff? Looks like the Pytest org switched from Travis to GitHub Actions.

**Comment by BeyondEvil** [comment]
> LGTM! @BeyondEvil need any help with the CI stuff? Looks like the Pytest org switched from Travis to GitHub Actions.

Thanks, waiting on someone to review my PR #36 after that this one can rebase and be merged.

**Review by BeyondEvil — APPROVED** [review]
LGTM!

As soon as my #36 is merged, we can rebase and merge this. 👍

**Comment by BeyondEvil** [comment]
Please rebase and we should be good to go! 🙏

**Comment by digitalorder** [comment]
> Please rebase and we should be good to go! 🙏

Rebased and fixed new tests on Windows. Github actions as CI are fun 👍

**Review by BeyondEvil — APPROVED** [review]
Awesome!

**Pull request merged**
